hibernate-core/src/main/java/org/hibernate/query/spi/AbstractSelectionQuery.java

```diff
@@ -374,8 +374,8 @@ public SelectionQuery<R> setHint(String hintName, Object value) {
 	}
 
 	@Override
-	public SelectionQuery<R> setEntityGraph(EntityGraph<R> graph, GraphSemantic semantic) {
-		applyGraph( (RootGraphImplementor<R>) graph, semantic );
+	public SelectionQuery<R> setEntityGraph(EntityGraph<? extends R> graph, GraphSemantic semantic) {
+		applyGraph( (RootGraphImplementor<?>) graph, semantic );
```

Comment:
Minor nitpick, just for consistency

```suggestion
		applyGraph( (RootGraphImplementor<? extends R>) graph, semantic );
```